content/en/platform/corda/5.0-dev-preview-1/tutorials/building-cordapp/c5-basic-cordapp-state-2.md

```diff
@@ -0,0 +1,212 @@
+  }
+```
+
+### Implement the `JsonRepresentable` interface to use the Corda RPC Client
+
+`JsonRepresentable` ensures the object can be serialized to JSON and called over RPC. Implement this interface in `BoardingTicket` so that it can be used with the Corda RPC Client.
+
+{{< note >}}
+There are several ways to return your parameters in a JSON string. The tutorial shows you one method, but you are not restricted to using this specific method in Corda.
+{{< /note >}}
+
+1. Create a data transfer object that encapsulates the data of your `BoardingTicket` state - `BoardingTicketDto`. Include the same variables as the `BoardingTicket` class (`description`, `marsExpress`, `owner`, and `daysUntilLaunch`) and mark all variable types as `String`.
+2. Create a function that instantiates the `BoardingTicketDto`.
+3. Create an override function that converts the `BoardingTicketDto` variables to JSON using the `toJson` method.
+
+Your code should now look like this:
+
+```kotlin
+package net.corda.missionMars.states
+
+import com.google.gson.Gson
+import net.corda.v5.application.identity.AbstractParty
+import net.corda.v5.application.identity.Party
+import net.corda.v5.application.utilities.JsonRepresentable
+import net.corda.v5.ledger.contracts.ContractState
+import java.time.LocalDate
+import java.util.*
+
+data class BoardingTicket(
+        var description : String, //Trip information
+        var marsExpress : Party, //Party selling the ticket
+        var owner: Party, //The party who exchanges the ticket for the voucher
+        var daysUntilLaunch: Int)
+    : ContractState, JsonRepresentable {
+
+    override val participants: List<AbstractParty> get() = listOf<AbstractParty>(marsExpress,owner)
+
+    fun toDto(): BoardingTicketDto {
+        return BoardingTicketDto(
+                description,
+                marsExpress.name.toString(),
+                owner.name.toString(),
+                daysUntilLaunch.toString()
+        )
+    }
+
+    override fun toJsonString(): String {
+        return Gson().toJson(this.toDto())
+    }
+}
+
+data class BoardingTicketDto(
+        var description : String, //Ticket information
+        var marsExpress : String, //Origin of the ticket
+        var owner: String, //The person who exchanges the ticket for the voucher
+        var daysUntilLaunch: String
+)
+```
+### Define a secondary constructor and helper method to change the ticket owner
+
+The `BoardingTicket` state is involved in two transactions. In the first transaction, Mars Express self-issues the `BoardingTicket`. The `marsExpress` party then fills both the `owner` and `marsExpress` fields of the transaction. In the second transaction, an ownership transfer occurs. This creates a new instance of the `BoardingTicket` state on the ledger, since you cannot change an immutable object. While states are immutable, the ledger as a whole is mutable as states are created and consumed.
```

Comment:
```suggestion
The `BoardingTicket` state is involved in two transactions. In the first transaction, Mars Express self-issues the `BoardingTicket`. The `marsExpress` party then fills both the `owner` and `marsExpress` fields of the transaction. In the second transaction, an ownership transfer occurs. Since you cannot change an immutable object, this creates a new instance of the `BoardingTicket` state on the ledger. While states are immutable, the ledger as a whole is mutable as states are created and consumed.
```